packages/server/BeaverTripleService/Server/EngineToBts/E2b.go

```diff
@@ -94,6 +95,70 @@ func (s *server) GetTriples(ctx context.Context, in *pb.GetTriplesRequest) (*pb.
 	}, nil
 }
 
+func (s *server) InitTripleStore(ctx context.Context, in *emptypb.Empty) (*emptypb.Empty, error) {
+	var reqIpAddrAndPort string
+	// ClientのIPアドレスを取得
+	if cs.Conf.WithEnvoy {
+		md, _ := metadata.FromIncomingContext(ctx)
+		port := strconv.FormatUint(uint64(cs.Conf.Port), 10)
+		reqIpAddrAndPort = fmt.Sprintf("%s:%s",md["x-forwarded-for"][0], port)
+	} else {
+		p, _ := peer.FromContext(ctx)
+		reqIpAddrAndPort = p.Addr.String()
+	}
+
+	partyId, err := GetPartyIdFromIp(reqIpAddrAndPort)
+	if err != nil {
+		return nil, err
+	}
+	logger.Infof("Ip %s, partyId: %d \n", reqIpAddrAndPort, partyId)
+
+	// TODO: read claims, and use these party information
+	claims, ok := ctx.Value("claims").(*jwt_types.Claim)
+	if ok {
+		logger.Infof("claims: %v\n", claims)
+	}
```

Comment:
It is redundant and I would like to see it summarized.

Reply:
I think that the part to get A should be a function, and it would be good to place claims in the interpreter.